clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -1669,18 +1669,28 @@ void MiscAPIRule::registerMatcher(MatchFinder &MF) {
   auto functionName = [&]() {
     return hasAnyName("cudaOccupancyMaxActiveBlocksPerMultiprocessor",
                       "cuOccupancyMaxActiveBlocksPerMultiprocessor",
-                      "cudaOccupancyMaxPotentialBlockSize");
+                      "cudaOccupancyMaxPotentialBlockSize",
+                      "cuGetExportTable");
   };
 
   MF.addMatcher(
       callExpr(callee(functionDecl(functionName()))).bind("FunctionCall"),
       this);
 }
 void MiscAPIRule::runRule(const MatchFinder::MatchResult &Result) {
-  const CallExpr *CE = getNodeAsType<CallExpr>(Result, "FunctionCall");
-  ExprAnalysis EA(CE);
-  emplaceTransformation(EA.getReplacement());
-  EA.applyAllSubExprRepl();
+  if (const CallExpr *CE = getNodeAsType<CallExpr>(Result, "FunctionCall")) {
+    auto FuncName = CE->getDirectCallee()->getNameInfo().getName().getAsString();
+
+    if (FuncName == "cuGetExportTable") {
+      auto Msg = MapNames::RemovedAPIWarningMessage.find(FuncName);
+      report(CE->getBeginLoc(), Diagnostics::API_REMOVED, false, MapNames::ITFName.at(FuncName), Msg->second);
+      emplaceTransformation(new ReplaceStmt(CE), "");
```

Comment:
why do we need handle this explicitly?
REMOVE_API_FACTORY_ENTRY_WITH_MSG should handle warning and removal automatically.

Reply:
Removed